test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/MSBuildProcessManager.cs

```diff
@@ -44,6 +42,17 @@ internal static class MSBuildProcessManager
                 processStartInfo.Arguments = $"msbuild {arguments}";
             }
 
+            var processResult = await RunProcessCoreAsync(processStartInfo, timeout);
+
+            return new MSBuildResult(project, processResult.FileName, processResult.Arguments, processResult.ExitCode, processResult.Output);
+        }
+
+        internal static Task<ProcessResult> RunProcessCoreAsync(
```

Comment:
Do we actually need any of these changes?

Reply:
I needed this so I could reuse all of the Thread, lock and output capturing goodness when running `dotnet rzc.dll shutdown -w`. Without this refactor, I am stuck with calling `dotnet msbuild ...`